packages/config/src/index.ts

```diff
@@ -23,7 +23,7 @@ class TruffleConfig {
     workingDirectory?: string,
     network?: any
   ) {
-    this._deepCopy = ["compilers"];
+    this._deepCopy = ["compilers", "mocha"];
```

Comment:
I'm not sure if there are any unexpected ramifications of putting this here. It looks like it only influences how `merge` works. Any other problems I didn't see?

Reply:
Couldn't see references beyond this file.